carma_wm_ctrl_ros2/include/carma_wm_ctrl_ros2/Geofence.hpp

```diff
@@ -0,0 +1,98 @@
+#pragma once
+/*
+ * Copyright (C) 2022 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+#include <lanelet2_core/primitives/Point.h>
+#include "GeofenceSchedule.hpp"
+#include <lanelet2_core/LaneletMap.h>
+#include <lanelet2_core/primitives/BasicRegulatoryElements.h>
+#include <lanelet2_core/primitives/Lanelet.h>
+#include <lanelet2_core/primitives/RegulatoryElement.h>
+#include <lanelet2_core/primitives/LaneletOrArea.h>
+#include <lanelet2_extension/regulatory_elements/DigitalSpeedLimit.h>
+#include <lanelet2_extension/regulatory_elements/DigitalMinimumGap.h>
+#include <lanelet2_extension/regulatory_elements/PassingControlLine.h>
+
+#include <boost/uuid/uuid.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <boost/uuid/uuid_generators.hpp>
+#include <carma_v2x_msgs/msg/traffic_control_message.hpp>
+#include <carma_v2x_msgs/msg/map_data.hpp>
+
+
+namespace carma_wm_ctrl
+{
+using namespace lanelet::units::literals;
+
+// Map Update Geofence Common Labels
+const std::string MAP_MSG_INTERSECTION = "MAP_MSG_INTERSECTION"; 
+const std::string MAP_MSG_TF_SIGNAL = "MAP_MSG_TF_SIGNAL";
+
+
+/**
+ * @brief An object representing a geofence use for communications with CARMA Cloud
+ *
+ * TODO: This is currently a place holder class which needs to be updated based on the final geofence specification
```

Comment:
We can probably remove this extremely old comment

Reply:
deleted